agent/rpc/peering/service.go

```diff
@@ -171,6 +171,11 @@ func (s *Server) GenerateToken(
 		return resp, err
 	}
 
+	// validate that this peer name is not being used as a PeerRole_DIALER
+	if err = s.validatePeerRole(req.PeerName, req.Partition, pbpeering.PeerRole_ACCEPTOR); err != nil {
```

Comment:
This makes the exact same state store call as `getExistingOrCreateNewPeerID` function below. Maybe there's a chance to fetch once and pass a `Peering` around to both functions as required?

Reply:
I made a small func to get an existing peering